File: rust/templates/api_extra/application_create.rs
/// Create the application with the given ID, or create a new one if it
/// doesn't exist yet.
pub async fn get_or_create(
    &self,
    application_in: ApplicationIn,
    options: Option<ApplicationCreateOptions>,
) -> Result<ApplicationOut> {
    let ApplicationCreateOptions { idempotency_key } = options.unwrap_or_default();

    crate::request::Request::new(http1::Method::POST, "/api/v1/app")
        .with_query_param("get_if_exists", "true".to_owned())
        .with_optional_header_param("idempotency-key", idempotency_key)
        .with_body_param(application_in)
        .execute(self.cfg)
        .await
}

File: ruby/templates/api_extra/application_create.rb
def get_or_create(application_in, options = {})
    options = options.transform_keys(&:to_s)
    path = "/api/v1/app"
    res = @client.execute_request(
    "POST",
    path,
    query_params:{
        "get_if_exists"=> true
    },
    headers: {
        "idempotency-key" => options["idempotency-key"]
    },
    body: application_in
    )
    ApplicationOut.deserialize(res)
end
